app/handlers/tests/test_handlers_resource.py

```diff
@@ -236,13 +265,16 @@ def test_sync_when_resource_exists_and_doesnt_need_update(
             }
         }
 
+        mock_api_client.get_resource.return_value = resource
+
         logger_mock = MagicMock()
         memo_mock = MagicMock()
         patch_mock = MagicMock()
         patch_mock.spec = {}
 
         twingate_resource_sync(
-            resource_spec.model_dump(by_alias=True, exclude="id"),
+            resource_spec.model_dump(by_alias=True),
```

Comment:
Since this is testing sync existing resource, we shouldn't exclude `id` here. Currently, the test will always pass because we skip the whole method ([here](https://github.com/Twingate/kubernetes-operator/blob/636c4961091cdd9e8deed6c69625a878ba417aa9/app/handlers/handlers_resource.py#L86)) if the `crd.id` is missing. 

In addition, the test setup is not complete, we should also return a resource in `get_resource`. This is fixed in `line 268` above